android/hello_sdl_android/src/main/java/com/sdl/hellosdlandroid/MainActivity.java

```diff
@@ -23,12 +22,14 @@ protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_main);
 
-
         if (BuildConfig.TRANSPORT.equals("MULTI") || BuildConfig.TRANSPORT.equals("MULTI_HB")) {
-            if (android.os.Build.VERSION.SDK_INT >= Build.VERSION_CODES.S && !checkPermission()) {
-                requestPermission();
-                return;
+            if (permissionsNeeded().length > 0) {
+                requestPermission(permissionsNeeded(), REQUEST_CODE);
+                if (checkBTPermission()) {
+                    return;
+                }
```

Comment:
With having an array with strings that contain our permission needs already it seems like we could check those with something like:

```java
for (String permission : permissionsNeeded) {
    if(Manifest.permission.BLUETOOTH_CONNECT.equals(permission)){
        return;
    }
}
```

I just don't know the efficiency of checking for the permission again, and it seems redundant. At the very least I would say your suggestions should be used as it's bit more concise.

Reply:
Another option would be to flip the methods for checking permission into returning if the permission is granted instead of currently returning if it is not granted, because that in itself is a big confusing as well.

Reply:
I agree that it was confusing especially with the naming of those methods, I flipped them and renamed to `hasPNPermission` and `hasBTPermission`. I wanted to not flip them and name them `needPNPermision` and `needBTPermission` since its checking the API level and then the permission, But `need` really isn't a proper java naming convention that I could find, so I added some JavDocs to them.